docs: set default file property in stackblitz.json

sonukapoor opened the pull request:
Previously, when no default file is defined in the `stackblitz.json`, stackblitz
will set it automatically. It used to be `main.ts` and nowadays it seems
to be `app.component.ts`. However, this behaviour can change in the future. To overcome
this, the stackblitz API allows specifying the file to open in the editor by passing
a query param. This PR sets the default file in the `stackblitz.json` through the
`file` property. If still no file is specified, the `builder.js` will try to default to
`app.component.html`, `app.component.ts` or `main.ts`.

Closes #22357

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:


## What is the current behavior?
<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->

Issue Number: #22357


## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

sonukapoor committed:
build(docs-infra): always specify a landing file for StackBlitz projects

The StackBlitz API (which we use to generate StackBlitz projects on the
fly, when a user clicks on a live example link in the docs) allows
specifying the file to open in the editor by passing a query param. If
no file is specified, StackBlitz opens a default one. In the past, it
used to be `main.ts` and nowadays it seems to be `app.component.ts`.

StackBlitz builder, the tool that we use to generate the StackBlitz
projects, allows specifying the primary file by setting the `file`
property in the corresponding `stackblitz.json`.

Previously, if the `file` property was not set, StackBlitz builder would
not specify a file, thus falling back on StackBlitz's default behavior.
This was not great, because the default behavior may change unexpectedly
is the future to something that is less useful for our users.

This commit change StackBlitz builder to always specify a primary file.
If no file is specified in `stackblitz.json`, the builder will look
for the first file that exists in the example from a list of predefined
files.

Partially addresses #22357.

gkalpak commented on aio/tools/stackblitz-builder/builder.js:
```diff
@@ -119,10 +119,16 @@ class StackblitzBuilder {
+      if (!selectedFile) {
+        throw new Error(`No default file found in ${config.basePath}`);
+      }
     }
```
To make this easier to follow, could you please move the logic to determine the `file` to a separate function?
Also, check the file's existence for `config.file` as well. Something like:

```js
function getPrimaryFile(config) {
  if (config.file) {
    if (!this._existsSync(path.join(config.basePath, config.file)) {
      throw new Error(`The specified primary file (${config.file}) does not exist in '${config.basePath}'.);
    }
    return config.file;
  } else {
    const defaultPrimaryFiles = ['src/app/app.component.html', 'src/app/app.component.ts', 'src/app/main.ts'];
    const primaryFile = defaultPrimaryFiles.find(fileName =>  this._existsSync(path.join(config.basePath, fileName)));

    if (!primaryFile) {
      throw new Error(`None of the default primary files (${defaultPrimaryFiles.join(', ')}) exists in '${config.basePath}'.);
    }

    return primaryFile;
  }
}
```

sonukapoor replied:
@gkalpak: Yes, the empty `main.2.ts` is intentional. The reason behind the file is that we are creating angular Stackblitz projects. If we include the default `main.ts`, then we will also need to include every file that is imported in the main.ts - e.g: `environment.ts`, `app.module`. It will also try to bootstrap, the application, which we do not want for the `aio\content\examples\docs-style-guide\second.stackblitz.json`. That file is used to showcase how we generate Stackblitz examples. Hope that makes sense. I have created a secondary commit for the builder as well.

gkalpak reviewed (changes requested):
Thx for working on this ❤️
I did an initial pass:

- Please move the changes to the builder to a commit of their own.
- Was adding an empty `aio/content/examples/docs-style-guide/src/main.2.ts` intentional? If not , please remove.

sonukapoor committed:
build(docs-infra): specify more relevant landing files for StackBlitz projects

The StackBlitz API (which we use to generate StackBlitz projects on the
fly, when a user clicks on a live example link in the docs) allows
specifying the file to open in the editor by passing a query param. If
no file is specified, StackBlitz opens a default one.

This commit specifies relevant landing files for some of the examples to
make it easier to engage with the examples more quickly, and to see what
the example is attempting to demonstrate.

Fixes #22357.

gkalpak reviewed (changes requested):
Nice work, @sonukapoor! (Sorry it took a while to review.)
I left a few minor comments. Also, please change the commit message of the first commit to something like:

> build(docs-infra): always specify a landing file for StackBlitz projects
> 
> The StackBlitz API (which we use to generate StackBlitz projects on the
> fly, when a user clicks on a live example link in the docs) allows
> specifying the file to open in the editor by passing a query param. If
> no file is specified, StackBlitz opens a default one. In the past, it
> used to be \`main.ts\` and nowadays it seems to be \`app.component.ts\`.
> 
> [StackBlitz builder][1], the tool that we use to generate the StackBlitz
> projects, allows specifying the primary file by setting the \`file\`
> property in the corresponding \`stackblitz.json\`.
> 
> Previously, if the \`file\` property was not set, StackBlitz builder would
> not specify a file, thus falling back on StackBlitz's default behavior.
> This was not great, because the default behavior may change unexpectedly
> is the future to something that is less useful for our users.
> 
> This commit change StackBlitz builder to always specify a primary file.
> If no file is specified in \`stackblitz.json\`, the builder will look
> for the first file that exists in the example from a list of predefined
> files.
> 
> Partially addresses #22357.
> 
> [1]: https://github.com/angular/angular/blob/9b9116c79/aio/tools/stackblitz-builder/builder.js

...and that of the second commit to something like:

> build(docs-infra): specify more relevant landing files for StackBlitz projects
> 
> The StackBlitz API (which we use to generate StackBlitz projects on the
> fly, when a user clicks on a live example link in the docs) allows
> specifying the file to open in the editor by passing a query param. If
> no file is specified, StackBlitz opens a default one.
> 
> This commit specifies relevant landing files for some of the examples to
> make it easier to engage with the examples more quickly, and to see what
> the example is attempting to demonstrate.
> 
> Fixes #22357.

sonukapoor commented:
@gkalpak I have made all the requested changes. Can you please re-review?

gkalpak reviewed (approved):
Awesome! Thx, @sonukapoor ❤️

gkalpak commented:
@caretaker: Global approval for the docs-only changes.
